storage/remote/queue_manager.go

```diff
@@ -45,6 +48,9 @@ const (
 
 	// Allow 30% too many shards before scaling down.
 	shardToleranceFraction = 0.3
+
+	invalidSegmentNumber = -1
```

Comment:
we should check the WAL package itself to see if these are set, and if not we should probably define them there